### PR TITLE
Class 03

### PR DESCRIPTION
**Pull request by smooto**

### CODE REVIEW

File: deprecated/Note.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    if(!Input.validate(action)) return console.log('Invalid input');
+
+    if(action.type === '--add') {
+      console.log('New note added: ' + action.payload);
+      return new Note(action);
+    }
+  }
+}
+
+module.exports = Note;
```
Comment (JMoneymaker) [inline deprecated/Note.js:19]: Include a newline at the end of each file.

File: index.js [hunk 2]
```diff
@@ -1,10 +1,26 @@
-const { Input } = require('./lib/Input');
-const { execute } = require('./lib/notes');
+require('dotenv').config();
+
+const mongoose = require('mongoose');
+const Note = require('./lib/models/Note');
+
```
Comment (JMoneymaker) [inline index.js:5]: Extra white space here.

File: index.js [hunk 3]
```diff
@@ -1,10 +1,26 @@
...
 
 const command = process.argv;
 
 // parse argv array and store result as action object
-const action = new Input(command).parse();
+const action = new Input(command);
 
 // execute appropriate action based on action object values
-execute(action);
+Note.execute(action).then(note => console.log(note));
```
Comment (JMoneymaker) [inline index.js:26]: You can chain a connection closure here as well.
.then(() => mongoose.connection.close());

File: deprecated/Note.test.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  });
+
+  it('doesn\'t return anything on invalid input', () => {
+    const action = {
+      type: '--goober'
+    };
+  
+    expect(Note.execute(action)).toBeFalsy();
+  });
+});
```
Comment (JMoneymaker) [inline deprecated/Note.test.js:23]: Include a newline at the end of each file.

File: lib/Input.js [hunk 5]
```diff
@@ -8,9 +8,9 @@ class Input {
...
 
-  validate() {
-    return this.type === '--add' && this.payload;
+  static validate(action) {
+    return action.type === '--add' && action.payload;
   }
 }
 
-module.exports = { Input };
+module.exports = Input;
```
Comment (JMoneymaker) [inline lib/Input.js:16]: Include a newline at the end of each file.

File: lib/models/Note.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    id: Math.random(),
+    body: payload
+  };
+
+  if(type === '--add' && payload) return this.create(newNote);
+
+  if(type === '--list') return this.find();
+};
+
+module.exports = mongoose.model('Note', noteSchema);
```
Comment (JMoneymaker) [inline lib/models/Note.js:27]: Include a newline at the end of each file.

File: deprecated/Note.js (hunk 1 above)
Comment (smooto) [reply deprecated/Note.js:19]: Oh, weird. I definitely changed that in my VSCode settings, and the newlines are present locally, so I'm not sure why they aren't persisting when I push. I'll look at my extensions/plugins/settings some more and see if I can get it figured out.

**Pull request merged**